api/lib/proxy.ts

```diff
@@ -12,38 +12,43 @@ import {
 } from '../configuration/references'
 
 import { http } from './http'
-//import * as log4jui from './log4jui'
 import { EnhancedRequest } from './models'
+import {exists} from './util'
 
-//const logger = log4jui.getLogger('proxy')
-
-export function setHeaders(req: EnhancedRequest) {
+export function setHeaders(req: express.Request) {
     const headers: any = {}
 
-    if (req.headers && req.headers['content-type']) {
-        headers['content-type'] = req.headers['content-type']
-    }
+    if (req.headers) {
 
-    if (req.headers && req.headers.accept) {
-        headers.accept = req.headers.accept
-    }
+        if (req.headers['content-type']) {
+            headers['content-type'] = req.headers['content-type']
+        }
 
-    if (req.headers && req.headers.experimental) {
-        headers.experimental = req.headers.experimental
-    }
+        if (req.headers.accept) {
+            headers.accept = req.headers.accept
+        }
 
-    if (req.auth && req.auth.token) {
-        headers.Authorization = `Bearer ${req.auth.token}`
-    }
+        if (exists(req, 'headers.experimental')) {
+            headers.experimental = req.headers.experimental
+        }
+
+        if (exists(req, 'headers.Authorization')) {
+            headers.Authorization = req.headers.Authorization
+        }
+
+        if (req.headers['user-roles'] && req.headers['user-roles'].length) {
+            headers['user-roles'] = req.headers['user-roles']
+        }
 
-    if (req.auth && req.auth.data && req.auth.data.roles && req.auth.data.roles.length > 0) {
-        headers['user-roles'] = req.auth.data.roles.join()
+        if (exists(req, 'headers.ServiceAuthorization')) {
+            headers.ServiceAuthorization = req.headers.ServiceAuthorization
+        }
     }
 
     return headers
 }
 
-export async function get(req: EnhancedRequest, res: express.Response, next: express.NextFunction) {
+export async function get(req: express.Request, res: express.Response, next: express.NextFunction) {
```

Comment:
can this be simplified to Request, Response, NextFunction?

Reply:
Yes it can, again a clean up task (tech debt for later)